internal/core/item_rendering.rs

```diff
@@ -478,8 +478,24 @@ pub trait ItemRenderer {
     }
 }
 
+/// After rendering an item, we cache the geometry and the transform it applies to
+/// children.
+#[derive(Clone)]
+pub struct CachedItemGeometryAndTransform {
+    /// The item's reported geometry, via [`ItemRc::geometry`].
+    pub geometry: LogicalRect,
+    /// The item's transform to apply to children, via [`ItemRc::children_transform`].
+    pub children_transform: ItemTransform,
+}
+
+impl From<(LogicalRect, ItemTransform)> for CachedItemGeometryAndTransform {
+    fn from((geometry, children_transform): (LogicalRect, ItemTransform)) -> Self {
+        Self { geometry, children_transform }
+    }
+}
+
 /// The cache that needs to be held by the Window for the partial rendering
-pub type PartialRenderingCache = RenderingCache<LogicalRect>;
+pub type PartialRenderingCache = RenderingCache<CachedItemGeometryAndTransform>;
```

Comment:
oh, that's a huge increase in the cache size :-/  do we really need to cache the transform? especially since it is going to be identity for most items.
Or is it acceptable?

Reply:
Good point. Let me change this to be an optional heap allocation. Then mcu pays the price of a null pointer and for everyone else it probably doesn't matter.